include/flatbuffers/base.h

```diff
@@ -153,6 +153,21 @@
   #pragma warning(disable: 4127) // C4127: conditional expression is constant
 #endif
 
+#ifndef FLATBUFFERS_STRING_VIEW
+  // Only provide string_view features if __has_include can be used to detect it
+  #if defined(__has_include)
+    // Check for std::string_view (in c++17)
+    #if __has_include(<string_view>) && (__cplusplus > 201402)
+      #include <string_view>
+      #define FLATBUFFERS_STRING_VIEW std::string_view
```

Comment:
I could maybe inject it into `namespace flatbuffers { using string_view = ...; }` or some such, to avoid the macro.

Reply:
(would still need a boolean `FLATBUFFERS_HAS_STRING_VIEW`, it just wouldn't do the typing this way.) I'm also not sure whether those `using`s are C++03-compliant (so it would have to be a `typedef`)?

Reply:
If it passes the CI, then its all good, so feel free to try that if it is cleaner. I'm also fine with the macro, since it is used just in implementation code.

Reply:
Done in eed85752, aea5d8a8